apps/s_client.c

```diff
@@ -798,6 +801,31 @@ static int new_session_cb(SSL *S, SSL_SESSION *sess)
     return 0;
 }
 
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    BIO *keylog = BIO_new_file(keylog_file, "a");
+    char buffer[256];
```

Comment:
I'm not sure it's a good idea for `ssl_log_secret` to add a newline.  What if the app callback places those records in a database?  Wouldn't a newline be unnecessary then?  It's better to leave it to the app to decide what to do with those lines and how to do it.

As for other programs, they should do it in whatever manner they see fit.  Not all programs are threaded, so concurrency isn't necessarely a problem for them.  On the other hand, if they need line buffering and use stdio, `setvbuf()` is the function to specify that with.